nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/adls/ListADLSFile.java

```diff
@@ -0,0 +1,289 @@
+                        directoryEntry.name.matches(fileFilter)
+                )
+                .map(directoryEntry -> {
+                    createAndTransferFlowFile(processSession, directoryEntry);
+                    return directoryEntry;
+                })
+                .max(Comparator.comparing(de -> de.lastModifiedTime));
+
+        Date currentRunLatestModifiedTimestamp;
+        try {
+            currentRunLatestModifiedTimestamp = optionalDirectoryEntry.get().lastModifiedTime;
+        } catch (NoSuchElementException e) {
+            getLogger().debug("There is no data to list. Yielding.");
+            processContext.yield();
+            return;
+        }
+
+        if(currentRunLatestModifiedTimestamp.after(latestTimestampListedDate)) {
+            processSession.commit();
+            getLogger().debug("Session commited");
+            try {
+                saveState(processContext, currentRunLatestModifiedTimestamp);
+            } catch (IOException e) {
+                getLogger().warn("Failed to save cluster-wide state. If NiFi is restarted, data duplication may occur", e);
+            }
+        }
+    }
+
+    private void saveState(ProcessContext processContext, final Date listedTimestamp) throws IOException {
+        final Map<String, String> updatedState = new HashMap<>(1);
+        updatedState.put(LISTING_TIMESTAMP_KEY, String.valueOf(listedTimestamp.getTime()));
+        processContext.getStateManager().setState(updatedState, Scope.CLUSTER);
+        getLogger().debug("New state saved with listed time: {}", new Object[] {listedTimestamp.toString()});
+    }
+
+    private void createAndTransferFlowFile(ProcessSession processSession, DirectoryEntry directoryEntry) {
+        final Map<String, String> attributes = createAttributes(directoryEntry);
+        FlowFile flowFile = processSession.create();
+        flowFile = processSession.putAllAttributes(flowFile, attributes);
+        processSession.transfer(flowFile, REL_SUCCESS);
+    }
+
+    private Map<String, String> createAttributes(final DirectoryEntry directoryEntry) {
+        final Map<String, String> attributes = new HashMap<>();
+        attributes.put(CoreAttributes.FILENAME.key(), directoryEntry.name);
+        Path fullNamePath = Paths.get(directoryEntry.fullName);
+        fullNamePath.toAbsolutePath().getParent().toString();
+        attributes.put(CoreAttributes.PATH.key(), fullNamePath.getParent().toString());
+        attributes.put(CoreAttributes.ABSOLUTE_PATH.key(), fullNamePath.toAbsolutePath().getParent().toString());
+        attributes.put("adls.owner", directoryEntry.user);
+        attributes.put("adls.group", directoryEntry.group);
+        attributes.put("adls.lastModified", String.valueOf(directoryEntry.lastModifiedTime));
+        attributes.put("adls.length", String.valueOf(directoryEntry.length));
+        attributes.put("adls.permissions", directoryEntry.permission);
+        if(directoryEntry.expiryTime != null)
```

Comment:
You need brackets. Please confirm the build is passing with ``-Pcontrib-check`` profile. Right now, you have the following checkstyle issues:
````
[WARNING] src/main/java/org/apache/nifi/processors/adls/ADLSAbstractProcessor.java[35] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - org.apache.nifi.processors.adls.ADLSConstants.*.
[WARNING] src/main/java/org/apache/nifi/processors/adls/ListADLSFile.java[22] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - org.apache.nifi.annotation.behavior.*.
[WARNING] src/main/java/org/apache/nifi/processors/adls/ListADLSFile.java[40] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - java.util.*.
[WARNING] src/main/java/org/apache/nifi/processors/adls/PutADLSFile.java[23] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - org.apache.nifi.annotation.behavior.*.
[WARNING] src/main/java/org/apache/nifi/processors/adls/PutADLSFile.java[48] (imports) AvoidStarImport: Using the '.*' form of import should be avoided - org.apache.nifi.processors.adls.ADLSConstants.*.
[WARNING] src/main/java/org/apache/nifi/processors/adls/PutADLSFile.java[223:9] (blocks) RightCurly: '}' should be on the same line.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[27:8] (imports) UnusedImports: Unused import - org.apache.nifi.util.MockFlowFile.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[34:8] (imports) UnusedImports: Unused import - java.io.File.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[35:8] (imports) UnusedImports: Unused import - java.io.FileInputStream.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[49] (sizes) LineLength: Line is longer than 200 characters (found 342).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[53] (sizes) LineLength: Line is longer than 200 characters (found 337).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestFetchADLSFile.java[54] (sizes) LineLength: Line is longer than 200 characters (found 333).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestListADLSFile.java[57] (sizes) LineLength: Line is longer than 200 characters (found 910).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestListADLSFile.java[58] (sizes) LineLength: Line is longer than 200 characters (found 1,934).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestListADLSFile.java[59] (sizes) LineLength: Line is longer than 200 characters (found 936).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestListADLSFile.java[60] (sizes) LineLength: Line is longer than 200 characters (found 1,569).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestListADLSFile.java[61] (sizes) LineLength: Line is longer than 200 characters (found 922).
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestPutADLSFile.java[34:8] (imports) UnusedImports: Unused import - java.io.ByteArrayInputStream.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestPutADLSFile.java[36:8] (imports) UnusedImports: Unused import - java.io.InputStream.
[WARNING] src/test/java/org/apache/nifi/processors/adls/TestPutADLSFile.java[37:8] (imports) UnusedImports: Unused import - java.nio.file.Paths.
````

Reply:
done.